packages/sdk/src/stores/google-datastore/store.ts

```diff
@@ -22,37 +25,61 @@ export default function <D>(kind: string, store: Datastore): SortedStore<string,
     });
   }
   async function has(id: string) {
-    try {
-      await get(id);
-      return true;
-    } catch (err) {
-      return false;
-    }
+    // horribly ineficient to actually query data to see if it exists, but cannot find a better way
+    return exists(await get(id));
   }
   async function del(id: string) {
     await store.delete(makeKey(id));
   }
   async function entries() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return [result[store.KEY].name, result];
+    }) as [string, D][];
   }
   async function values() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results;
   }
+  // all this does is map over full entries to return the id of the entry. have not found a better way to query this.
```

Comment:
We might be able to use [select](https://googleapis.dev/nodejs/datastore/latest/Query.html#select) to only pull the key? But not sure if that works for keys

Reply:
yea this might work. ill give it a try

Reply:
updated this to select `__key__`, which is some special identifier for just keys